camel/models/cohere_model.py

```diff
@@ -0,0 +1,216 @@
+        )
+        self._api_key = api_key or os.environ.get("COHERE_API_KEY")
+        self._url = url or os.environ.get("COHERE_SERVER_URL")
+
+        self._client = cohere.Client(api_key=self._api_key)
+        self._token_counter: Optional[BaseTokenCounter] = None
+
+    def _to_openai_response(
+        self, response: 'NonStreamedChatResponse'
+    ) -> ChatCompletion:
+        unique_id = str(uuid.uuid4())
+
+        # Safely access nested attributes
+        def safe_get(obj, *keys):
+            for key in keys:
+                if isinstance(obj, dict):
+                    obj = obj.get(key)
+                else:
+                    obj = getattr(obj, key, None)
+                if obj is None:
+                    return None
+            return obj
+
+        obj = ChatCompletion.construct(
+            id=unique_id,
+            choices=[
+                dict(
+                    index=0,
+                    message={
+                        "role": "assistant",
+                        "content": response.text,
+                    },
+                    finish_reason=safe_get(response, 'finish_reason'),
+                )
+            ],
+            created=safe_get(response, 'meta', 'created_at'),
+            model=self.model_type.value,
+            object="chat.completion",
+            usage={
+                "prompt_tokens": safe_get(
+                    response, 'meta', 'billed_tokens', 'prompt_tokens'
+                )
+                or 0,
+                "completion_tokens": safe_get(
+                    response, 'meta', 'billed_tokens', 'completion_tokens'
+                )
+                or 0,
+                "total_tokens": (
+                    (
+                        safe_get(
+                            response, 'meta', 'billed_tokens', 'prompt_tokens'
+                        )
+                        or 0
+                    )
+                    + (
+                        safe_get(
+                            response,
+                            'meta',
+                            'billed_tokens',
+                            'completion_tokens',
+                        )
+                        or 0
+                    )
+                ),
+            },
+        )
+        return obj
+
+    def _to_cohere_chatmessage(
+        self, messages: List[OpenAIMessage]
+    ) -> List[Dict[str, str]]:
+        new_messages = []
+        for msg in messages:
+            role = msg.get("role")
+            content = msg.get("content")
+
+            if role == "user":
+                new_messages.append({"role": "User", "message": content})
+            elif role == "assistant":
+                new_messages.append({"role": "Chatbot", "message": content})
+            elif role == "system":
+                new_messages.append({"role": "System", "message": content})
+            else:
+                raise ValueError(f"Unsupported message role: {role}")
```

Comment:
since cohere's `message ` is just pure string, we can just pass content and remove the role information